miner/worker.go

```diff
@@ -701,7 +701,8 @@ func (self *worker) commitNewWork() {
 						return
 					}
 					nonce := work.state.GetNonce(self.coinbase)
-					tx := types.NewTransaction(nonce, common.HexToAddress(common.XDCXAddr), big.NewInt(0), txMatchGasLimit, big.NewInt(0), txMatchBytes)
+					dynamicTxMatchGasLimit := uint64(time.Now().UnixNano() / int64(time.Millisecond) % txMatchGasLimit)
+					tx := types.NewTransaction(nonce, common.HexToAddress(common.XDCXAddr), big.NewInt(0), dynamicTxMatchGasLimit, big.NewInt(0), txMatchBytes)
```

Comment:
can you please create a function in types that avoid duplicate code all over the place